src/simpleDetFunc.c

```diff
@@ -376,21 +378,21 @@ int simpleDetFunctionCalculation(int correntropyWinSize, int interval,
 	 * function that computes correntropy. Add correntropyWinSize zeros 
 	 * will be sufficient.
 	 */
-	bufferLength = dataLength + correntropyWinSize;
+	bufferLength = (numWindows-1)*interval + 2*correntropyWinSize + 2;
```

Comment:
just wondering why this change was made and how you determined this new bufferlength. updating the comment above this line is probably a good idea.

Reply:
I calculated it based on the logic in pSMContribution. I had updated the documentation to state that the buffer argument was assumed to have a length of at least `(numWindows-1)*interval + 2*correntropyWinSize + 2` and confirmed that it was accurate with a test. 
The old length was just a conservative guess at the required length.

In the final commit added to the pull request, I updated the comment.